### PR TITLE
Fix the behavior of `gr.LoginButton` locally and on Spaces

### PR DESCRIPTION
**Pull request by abidlabs**
Closes: https://github.com/gradio-app/gradio/issues/9650

Also changes the fixes the styling of the `gr.LoginButton` as it was not showing up correctly

Test here: https://huggingface.co/spaces/abidlabs/login5?logs=build

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
@abidlabs , does this also cover non-space logins? The login does not seem to work locally either.

**Comment by abidlabs** [comment]
Hi @davidberenstein1957 yup it will

File: js/button/shared/Button.svelte [hunk 1]
```diff
@@ -190,8 +190,14 @@
 	}
 
 	.huggingface {
-		border: var(--button-border-width) solid
-			var(--button-secondary-border-color);
-		background: var(--background-fill-primary);
+		background: black;
```
Comment (abidlabs) [inline js/button/shared/Button.svelte:193]: The "huggingface" variant should _not_ change depending on the theme, in order to preserve the Hugging Face branding

**Comment by abidlabs** [comment]
Should be ready for review, cc @apolinario

**Review by freddyaboulton — COMMENTED** [review]
Python changes look good! 

Though the white button on dark mode looks off to me

image

**Comment by abidlabs** [comment]
Thanks for the review @freddyaboulton, update the style, [see screenshots here](https://huggingface.slack.com/archives/C03BHNJCBC4/p1729031740633239?thread_ts=1729028330.065169&cid=C03BHNJCBC4).

**Review by freddyaboulton — APPROVED** [review]

**Comment by freddyaboulton** [comment]
Thanks for the fix!

**Review by hysts — APPROVED** [review]
LGTM!

**Comment by abidlabs** [comment]
Thanks @freddyaboulton @hysts @aliabid94 for the review and feedback!

**Review by Wauplin — COMMENTED** [review]
Much more robust, thanks @abidlabs for fixing this!

**Pull request merged**